src/simulation.hpp

```diff
@@ -1224,8 +1224,8 @@ template <typename problem_t> void AMRSimulation<problem_t>::evolve()
 			break;
 		}
 
-		if (maxWalltime_ > 0 && getWalltime() > 0.9 * maxWalltime_) {
-			// we have exceeded 90% of maxWalltime_
+		if (maxWalltime_ > 0 && getWalltime() > std::max(0.9 * maxWalltime_, maxWalltime_ - 300)) {
```

Comment:
This should now sensibly handle the case where the walltime is less than 5 minutes.